sqfvm-cpp/genericcmds.cpp

```diff
@@ -518,6 +518,124 @@ namespace
 		auto r = right->data<arraydata>();
 		return std::make_shared<value>(r->operator std::vector<std::shared_ptr<sqf::value>, std::allocator<std::shared_ptr<sqf::value>>>());
 	}
+	std::shared_ptr<value> plus_array_array(virtualmachine* vm, std::shared_ptr<value> left, std::shared_ptr<value> right)
+	{
+		// create a copy of left array
+		auto result = std::make_shared<value>(left->as_vector());
+		result->data<arraydata>()->extend(right->as_vector());
+		return result;
+	}
+
+	bool value_equals_casesensitive(std::shared_ptr<value> a, std::shared_ptr<value> b)
+	{
+		// TODO: put this into value class? make use of it elsewhere
+		if (a->dtype() == type::STRING && a->dtype() == b->dtype())
+		{
+			return a->as_string() == b->as_string();
+		}
+		else
+		{
+			return a->equals(b);
+		}
+	}
+
+	std::shared_ptr<value> minus_array_array(virtualmachine* vm, std::shared_ptr<value> left, std::shared_ptr<value> right)
+	{
+		// TODO: optimize (e.g. build a hash based set of the right array and check against it)
```

Comment:
not in scope of sqfvms goal (yet), so more a long time goal then everything else 😄